src/lib.rs

```diff
@@ -3,86 +3,114 @@ extern crate dotenv_codegen;
 
 use arma_rs::{arma, Extension};
 use chrono::prelude::{DateTime, Utc};
-use env_logger::{Builder, Target};
-use futures::executor::block_on;
+use env_logger::Target;
 use log::LevelFilter;
 use mongodb::{
     bson::{doc, Document},
-    options::ClientOptions,
-    options::UpdateOptions,
-    Client, Database,
+    options::{ClientOptions, UpdateOptions},
+    sync::{Client, Database},
 };
 use once_cell::sync::OnceCell;
-use std::{env, thread, time::SystemTime};
+use std::{env, sync::RwLock, thread, time::SystemTime};
 
-static MONGODB: OnceCell<Database> = OnceCell::new();
+static MONGODB: OnceCell<RwLock<Option<Database>>> = OnceCell::new();
 
 #[arma]
 fn init() -> Extension {
-    let mut builder = Builder::from_default_env();
+    let mut builder = env_logger::Builder::from_default_env();
     builder.target(Target::Stdout);
     builder.filter_level(LevelFilter::Info);
     builder.init();
-
+    // We unwrap as the cell should not be initialized
+    MONGODB.set(RwLock::new(None)).unwrap();
     Extension::build().command("log", log).finish()
 }
 
-async fn connect() {
-    if MONGODB.get().is_some() {
-        log::warn!(target: "fp_extension", "Connection to DB already present!");
-        return;
-    }
-
-    log::info!(target: "fp_extension", "Connecting to DB!");
+fn connect() -> Result<(), ()> {
+    // We unwrap as the cell should be initialized by now.
+    let lock = MONGODB.get().unwrap();
 
-    let _url = env::var("FP_EXTENSION_MONGO_DB_URL")
-        .unwrap_or_else(|_| dotenv!("FP_EXTENSION_MONGO_DB_URL").to_string());
-    let _db_name = env::var("FP_EXTENSION_MONGO_DB_DBNAME")
-        .unwrap_or_else(|_| dotenv!("FP_EXTENSION_MONGO_DB_DBNAME").to_string());
+    {
+        // In this scope we hold shared access to the lock.
+        let read_l = lock.read().unwrap();
+        if read_l.as_ref().is_some() {
+            log::warn!(target: "fp_extension", "Connection to DB already present!");
```

Comment:
can this be changed to debug, it would say on every log that a connection is already present in the a3 console

Old implementation did it a bit differently, the log would only appear if the connect func would be called again.